src/emulator/auth/widget_ui.ts

```diff
@@ -142,15 +142,23 @@ function redirectToIos(authEvent) {
 
 // DOM logic
 
-document.querySelectorAll('.js-provider-id').forEach(function(e) {e.textContent = providerId});
+var formattedProviderId = providerId[0].toUpperCase() + providerId.substring(1);
+document.querySelectorAll('.js-provider-id').forEach(function(e) {
+  e.textContent = formattedProviderId;
+});
+
 var reuseAccountEls = document.querySelectorAll('.js-reuse-account');
-[].forEach.call(reuseAccountEls, function (el) {
-  var urlEncodedIdToken = el.dataset.idToken;
-  el.addEventListener('click', function (e) {
-    e.preventDefault();
-    finishWithUser(urlEncodedIdToken);
+if (reuseAccountEls.length) {
+  [].forEach.call(reuseAccountEls, function (el) {
```

Comment:
Out of curiosity, why are you doing
```js
[].forEach.call(reuseAccountEls, ...
```
instead of
```js
reuseAccountEls.forEach(...
```

Reply:
Apparently I have the memory of a goldfish

Reply:
consider using `Array.from(reuseAccountEls)`, (or even `[...reuseAccountEls]`) for better readability

Reply:
Sorry but no ES6 for now. In the long term we may even get a transpiler, but not now.